docs/pages/management/infrastructure-as-code/terraform-starter/rbac.mdx

```diff
@@ -0,0 +1,537 @@
+
+   - [AD FS](../../../access-controls/sso/adfs.mdx#step-13-configure-adfs)
+   - [GitLab](../../../access-controls/sso/gitlab.mdx#step-13-configure-gitlab)
+   - [Google
+     Workspace](../../../access-controls/sso/google-workspace.mdx#step-14-configure-google-workspace)
+   - [OneLogin](../../../access-controls/sso/one-login.mdx#step-13-create-teleport-application-in-onelogin)
+   - [Azure
+     AD](../../../access-controls/sso/azuread.mdx#step-13-configure-azure-ad)
+   - [Okta](../../../access-controls/sso/okta.mdx#step-24-configure-okta)
+
+1. Configure the redirect URL (for OIDC) or assertion consumer service (for SAML):
+
+   <Tabs>
+   <TabItem label="OIDC">
+
+   Set `oidc_redirect_url` to
+   `https://example.teleport.sh:443/v1/webapi/oidc/callback`, replacing
+   `example.teleport.sh` with the domain name of your Teleport cluster. 
+
+   Ensure that `oidc_redirect_url` matches match the URL you configured with
+   your IdP when registering your Teleport cluster as a relying party.
+
+   </TabItem>
+   <TabItem label="SAML">
+
+   Set `saml_acs` to `https://example.teleport.sh:443/v1/webapi/saml/acs`,
+   replacing `example.teleport.sh` with the domain name of your Teleport
+   cluster. 
+
+   Ensure that `saml_acs` matches the URL you configured with your IdP when
+   registering your Teleport cluster as a relying party.
+
+   </TabItem>
+   </Tabs>
+
+1. After you register Teleport as a relying party, your identity provider will
+   print information that you will use to configure the authentication
+   connector. Fill in the information depending on your provider type:
+
+   <Tabs>
+   <TabItem label="OIDC">
+
+   Fill in the `oidc_client_id` and `oidc_secret` with the client ID and secret
+   returned by the IdP.
+
+   </TabItem>
+   <TabItem label="SAML">
+
+   Assign `saml_entity_descriptor` to the contents of the XML document that
+   contains the SAML entity descriptor for the IdP.
+
+   </TabItem>
+   </Tabs>
+
+1. Assign `teleport_domain` to the domain name of your Teleport Proxy Service,
+   with no scheme or path, e.g., `example.teleport.sh`. The child module uses
+   this to configure WebAuthn for local users. This way, you can authenticate as
+   a local user as a fallback if you need to troubleshoot your single sign-on
+   authentication connector.
+
+1. Configure role mapping for your authentication connector. When a user
+   authenticates to Teleport through your organization's IdP, Teleport assigns
+   roles to the user based on your connector's role mapping configuration:
+
+   <Tabs>
+   <TabItem label="OIDC">
+
+   In this example, users with a `group` claim with the `developers` value
+   receive the `dev_access` role, while users with a `group` claim with the
+   value `admins` receive the `prod_reviewer` role:
+
+   ```hcl
+        oidc_claims_to_roles = [
+          {
+            claim = "group"
+            value = "developers"
+            roles = [
+              module.dev_role.role_name
```

Comment:
Thank you so much for referencing the name dynamically instead of putting a string, we got this wrong in many of our own examples.

This is very important as TF needs this to understand there is a dependency between both resources. This ensures you cannot try to delete a role which is still in the mappings.